Extensions/LinkedObjects/linkedobjects.ts

```diff
@@ -112,30 +216,125 @@ namespace gdjs {
+          objectsLists,
+          obj,
+          // Can't know if it's called from an event function or not
+          // Object names will have to be checked
+          true
+        );
+      };
+
+      const doPickObjectsLinkedTo = function (
+        runtimeScene: gdjs.RuntimeScene,
+        objectsLists: Hashtable<gdjs.RuntimeObject[]>,
+        obj: gdjs.RuntimeObject,
+        isEventsFunction: boolean
       ) {
         if (obj === null) {
           return false;
         }
-        const linkedObjects = LinksManager.getManager(
+        const linkedObjectMap = LinksManager.getManager(
           runtimeScene
-        ).getObjectsLinkedWith(obj);
-        return gdjs.evtTools.object.pickObjectsIf(
-          gdjs.evtTools.linkedObjects._objectIsInList,
-          objectsLists,
-          false,
-          linkedObjects
-        );
+        )._getMapOfObjectsLinkedWith(obj);
+
+        let pickedSomething = false;
+        for (const contextObjectName in objectsLists.items) {
+          if (objectsLists.containsKey(contextObjectName)) {
+            const pickedObjects = objectsLists.items[contextObjectName];
+
+            if (pickedObjects.length === 0) {
+              continue;
+            }
+
+            // Find the object names in the scene
+            const temporaryObjectNames = gdjs.staticArray2(
+              gdjs.evtTools.linkedObjects.pickObjectsLinkedTo
+            );
+            temporaryObjectNames.length = 0;
+            if (isEventsFunction) {
+              // For functions, objects may be a flattened group
+              for (const pickedObject of pickedObjects) {
+                if (temporaryObjectNames.indexOf(pickedObject.getName()) < 0) {
+                  temporaryObjectNames.push(pickedObject.getName());
+                }
+              }
+            } else {
+              temporaryObjectNames.push(contextObjectName);
+            }
+
+            let objectCount = 0;
+            let linkedObjectExists = false;
+            for (const objectName of temporaryObjectNames) {
+              objectCount += runtimeScene.getObjects(objectName).length;
+              linkedObjectExists =
+                linkedObjectExists || linkedObjectMap.has(objectName);
+            }
+
+            if (linkedObjectExists) {
+              if (pickedObjects.length === objectCount) {
+                // All the objects were picked, there is no need to make an intersection.
+                pickedObjects.length = 0;
+                for (const objectName of temporaryObjectNames) {
+                  if (linkedObjectMap.has(objectName)) {
+                    const linkedObjects = linkedObjectMap.get(objectName)!;
+
+                    pickedSomething =
+                      pickedSomething || linkedObjects.length > 0;
+                    pickedObjects.push.apply(pickedObjects, linkedObjects);
+                  }
+                }
+              } else {
+                const temporaryObjects = gdjs.staticArray(
```

Comment:
Same, avoid "temporary".
Instead, I think this represents the objects that are picked *and* linked to `obj`. So let's call this `pickedAndLinkedObjects`? (i.e: that's the intersection that we're doing here)